internal/validation/v2alpha1/rules.go

```diff
@@ -48,6 +49,7 @@ func validateRules(ctx context.Context, client client.Client, parentAttributePat
 			problems = append(problems, extAuthFailures...)
 		}
 
+		problems = append(problems, validatePath(ruleAttributePath, rule.Path)...)
```

Comment:
If you return slice of failures with only one failure, refactor it to append only one failure

Reply:
This is easier to handle as the function does not need to return a pointer to validation.Failure, as slices are pointers by default. In that context what would you prefer: return slice, or return a pointer and check if the return is nil?